indico/modules/events/editing/client/js/editing/timeline/selectors.js

```diff
@@ -8,135 +8,45 @@
 import _ from 'lodash';
 import {createSelector} from 'reselect';
 
-import {Translate} from 'indico/react/i18n';
+import {FinalRevisionState, InitialRevisionState} from '../../models';
 
-import {InitialRevisionState, FinalRevisionState} from '../../models';
+import {getRevisionTransition} from './util';
 
-function shouldCreateNewRevision({initialState, finalState}) {
-  return (
-    initialState.name === InitialRevisionState.needs_submitter_confirmation ||
-    (initialState.name === InitialRevisionState.ready_for_review &&
-      ![
-        FinalRevisionState.accepted,
-        FinalRevisionState.rejected,
-        FinalRevisionState.replaced,
-      ].includes(finalState.name))
-  );
+// This method defines each revision as a block
+// with a label referring to its previous state transition
+export function processRevisions(revisions) {
```

Comment:
could you move this whole function into `util.js` since it's not a selector?

Reply:
I agree, always hated to see this here.